gerrychain/tree.py

```diff
@@ -151,11 +159,37 @@ def part_nodes(start):
                                 queue.append(c)
             return nodes
 
-        if abs(tree_pop - h.ideal_pop) <= h.ideal_pop * h.epsilon:
-            cuts.append(Cut(edge=(node, pred[node]), subset=part_nodes(node)))
-        elif abs((total_pop - tree_pop) - h.ideal_pop) <= h.ideal_pop * h.epsilon:
-            cuts.append(Cut(edge=(node, pred[node]),
-                            subset=set(h.graph.nodes) - part_nodes(node)))
+        is_balanced_A = abs(tree_pop - h.ideal_pop) <= h.ideal_pop * h.epsilon
+        is_balanced_B = abs((total_pop - tree_pop) - h.ideal_pop) <= h.ideal_pop * h.epsilon
+
+        parent = pred[node]
+        if region_weights is not None:
+            region_cols = [tup[0] for tup in region_weights]
+            # score function to prefer dividing higher ranked regions. three regions -> [4, 2, 1]
+            # an edge that splits our first- and last-ranked regions would get a score of 4 + 1 = 5
+            # if node corresponds to a balance cut and our split score is as good or better than our
+            # best previously seen split score, add this cut to our list of potential balance cuts
+            node_split_score = 0
+            for i, region_col in enumerate(region_cols):
+                if h.graph.nodes[parent][region_col] != h.graph.nodes[node][region_col]:
+                    node_split_score += 2 ** (len(region_cols) - i - 1)
```

Comment:
If we're reading this right, the weights contributed by the regions depends on the _ordering_ of `region_weights`, not on the right values of the 2-tuples in `region_weights`. This is confusing because it's undocumented—as a user, there's no _a priori_ reason for me to believe  `region_weights=[("COUNTYFP20", 1), ("TOWN", 1)]` will yield distinct behavior from `region_weights=[("TOWN", 1), ("COUNTYFP20", 1)]`.